Utils/SerDes/SerDes.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include <fcntl.h>
+#include <string>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <cstring>
+
+template<class T>
+void serialization(const std::string& file, const T* obj ) {
+    int fd = open(file.c_str(), O_WRONLY | O_CREAT, S_IRUSR | S_IWUSR);
+
+    auto bytes = reinterpret_cast<const char*>(obj);
+    write(fd, bytes, sizeof(T));
+
+    close(fd);
+}
+
+template<class T>
+void deserialization(const std::string& file, T* obj) {
```

Comment:
The use of a pointer forces you to check it against `nullptr` which you didn't do. Make it `T&`.